src/rust/ensogl/lib/components/src/file_browser/column.rs

```diff
@@ -0,0 +1,442 @@
+
+use crate::prelude::*;
+
+use crate::list_view;
+use crate::file_browser::{ModelWithFrp, icons};
+use crate::file_browser::model::*;
+use crate::list_view::ListView;
+use crate::file_browser::icons::DynamicIcon;
+use crate::shadow;
+use crate::file_browser;
+
+use enso_frp as frp;
+use ensogl_core::application::Application;
+use ensogl_core::{display, Animation};
+use ensogl_core::display::shape::*;
+use std::path::PathBuf;
+use ensogl_text as text;
+use ensogl_core::data::color;
+use ensogl_theme as theme;
+
+
+// =================
+// === Constants ===
+// =================
+
+const ENTRY_PADDING_LEFT   : f32 = 6.0;
+const ICON_TO_LABEL        : f32 = 5.0;
+const LABEL_TO_ARROW       : f32 = 6.0;
+const ENTRY_PADDING_RIGHT  : f32 = 2.0;
+const STATE_LABEL_PADDING  : f32 = 20.0;
+const STATE_LABEL_OFFSET_Y : f32 = 20.0;
+
+fn entry_color_normal() -> color::Rgba {
+    color::Rgba(0.341,0.341,0.341,1.0)
+}
+
+fn entry_color_focused() -> color::Rgba {
+    color::Rgba::white()
+}
+
+fn state_label_color() -> color::Rgba {
+    color::Rgba(0.5,0.5,0.5,1.0)
+}
+
+
+
+// ==============
+// === Shadow ===
+// ==============
+
+mod column_shadow {
+    use super::*;
+
+    pub const SHADOW_PX:f32 = 10.0;
+
+    ensogl_core::define_shape_system! {
+        (style:Style,opacity:f32) {
+            let background = HalfPlane().rotate(90.0_f32.to_radians().radians());
+            let color      = style.get_color(theme::application::file_browser::background);
+            let background = background.fill(color);
+            let shadow     = shadow::from_shape_with_alpha((&background).into(),&opacity,style);
+
+            (shadow + background).into()
+        }
+    }
+}
+
+
+// =========================
+// === ListEntryProvider ===
+// =========================
+
+#[derive(Debug)]
+struct ListEntry {
+    display_object : display::object::Instance,
+    label          : text::Area,
+    icon           : Box<dyn DynamicIcon>,
+    arrow          : Option<super::icons::Arrow>,
+}
+
+impl ListEntry {
+    fn new(app:&Application, entry:Entry) -> Self {
+        let logger         = Logger::new("ListEntry");
+        let layers         = &app.display.scene().layers;
+        let display_object = display::object::Instance::new(&logger);
+
+        let label = text::Area::new(app);
+        display_object.add_child(&label);
+        label.set_position_x(ENTRY_PADDING_LEFT+icons::ICON_SIZE +ICON_TO_LABEL);
```

Comment:
strange spacing between operands. 
```suggestion
        label.set_position_x(ENTRY_PADDING_LEFT + icons::ICON_SIZE + ICON_TO_LABEL);
```